test_conformance/images/kernel_image_methods/test_1D_buffer.cpp

```diff
@@ -0,0 +1,282 @@
+//
+// Copyright (c) 2023 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "../testBase.h"
+#include <CL/cl.h>
+
+
+struct image_kernel_data
+{
+    cl_int width;
+    cl_int channelType;
+    cl_int channelOrder;
+    cl_int expectedChannelType;
+    cl_int expectedChannelOrder;
+};
+
+static const char *methodTest1DImageKernelPattern =
+    "typedef struct {\n"
+    "    int width;\n"
+    "    int channelType;\n"
+    "    int channelOrder;\n"
+    "    int expectedChannelType;\n"
+    "    int expectedChannelOrder;\n"
+    " } image_kernel_data;\n"
+    "__kernel void sample_kernel( %s image1d_buffer_t input, __global "
+    "image_kernel_data *outData )\n"
+    "{\n"
+    "   outData->width = get_image_width( input );\n"
+    "   outData->channelType = get_image_channel_data_type( input );\n"
+    "   outData->channelOrder = get_image_channel_order( input );\n"
+    "\n"
+    "   outData->expectedChannelType = %s;\n"
+    "   outData->expectedChannelOrder = %s;\n"
+    "}";
+
+static int test_get_1Dimage_buffer_info_single(cl_context context,
+                                               cl_command_queue queue,
+                                               image_descriptor *imageInfo,
+                                               MTdata d, cl_mem_flags flags)
+{
+    int error = 0;
+
+    clProgramWrapper program;
+    clKernelWrapper kernel;
+    clMemWrapper image, outDataBuffer, buffer;
+    char programSrc[10240];
+
+    image_kernel_data outKernelData;
+
+    // Generate some data to test against
+    BufferOwningPtr<char> imageValues;
+    generate_random_image_data(imageInfo, imageValues, d);
+
+    // Construct testing source
+    if (gDebugTrace)
+        log_info(" - Creating 1D image %d ...\n", (int)imageInfo->width);
+
+    buffer = clCreateBuffer(
```

Comment:
The size passed to `clCreateBuffer` and the one in the call to `log_error` are different. The correct value that should be passed to both is `imageInfo->rowPitch`.

Reply:
Shouldn't `rowPitch` always equal `width * pixel_size`?

But I agree, it would be better to remove the use of `imageInfo->width * get_pixel_size(imageInfo->format)` in favour of `imageInfo->rowPitch`.

Reply:
It should, but before the call to this function there is:
`imageInfo.rowPitch += extraWidth;`

Reply:
Then the caller might also need to be fixed either way, right?

Reply:
Sorry, I didn't dig into what the test is trying to do but my general comment is that whatever we pass in as the image row pitch for creating the image, should be the same value we use for creating the buffer (size of the buffer is required to be >= row pitch in this case), and the same value that we log.